drivers/volume/portworx/portworx.go

```diff
@@ -379,6 +312,49 @@ func (p *portworx) initPortworxClients() error {
 	return err
 }
 
+// 	tokenGenerator generates authorization token for system.admin
+//	when shared secret is not configured authz token is empty string
+//	this let Openstorage API clients be bootstrapped with no authorization (by accepting empty token)
+func (p *portworx) tokenGenerator() (string, error) {
+	if len(p.jwtSharedSecret) == 0 {
+		return "", nil
+	}
+
+	claims := &auth.Claims{
+		Issuer: "stork.openstorage.io",
+		Name:   "Stork",
+
+		// Unique id for stork
+		// this id must be unique across all accounts accessing the px system
+		Subject: "stork.openstorage.io." + p.id,
```

Comment:
we will need @ggriffiths  changes for 2.6 as well? https://github.com/libopenstorage/stork/pull/690/files#diff-0aa9aa57d00bdd00603aae645329fb04R100

Reply:
@ggriffiths Do we need to cherry-pic that change too? Merging this PR for now

Reply:
@disrani-px You do not need that fix. The porx system is backwards compatible.